lib/azure/armrest/armrest_collection.rb

```diff
@@ -4,6 +4,7 @@ module Azure
   module Armrest
     class ArmrestCollection < Array
       attr_accessor :continuation_token
+      attr_accessor :response_headers
```

Comment:
Upon further review, maybe this would be easier, since we have a bunch of custom list methods scattered throughout.